main/vstring.c

```diff
@@ -115,8 +108,8 @@ extern vString *vStringNewInit (const char *const s)
 static void stringCat (
 		vString *const string, const char *const s, const size_t length)
 {
-	while (string->length + length + 1 >= string->size)
-		vStringAutoResize (string);
+	if (string->length + length + 1 > string->size)
+		vStringResize (string, string->length + length + 1);
```

Comment:
that's better IMO: it requests the exact size it wants, and lets `vStringResize()` do what it wants.